src/devices/machine/74259.h

```diff
@@ -0,0 +1,173 @@
+
+***********************************************************************
+                              ____   ____
+                      A0   1 |*   \_/    | 16  Vcc
+                      A1   2 |           | 15  /C
+                      A2   3 |           | 14  /E
+                      Q0   4 |  SN74259  | 13  D
+                      Q1   5 |  F  9334  | 12  Q7
+                      Q2   6 |           | 11  Q6
+                      Q3   7 |           | 10  Q5
+                     GND   8 |___________|  9  Q4
+
+                              ____   ____
+                      Q7   1 |*   \_/    | 16  Vdd
+                   RESET   2 |           | 15  Q6
+                    DATA   3 |           | 14  Q5
+           WRITE DISABLE   4 |           | 13  Q4
+                      A0   5 |  CD4099B  | 12  Q3
+                      A1   6 |           | 11  Q2
+                      A2   7 |           | 10  Q1
+                     Vss   8 |___________|  9  Q0
+
+**********************************************************************/
+
+#pragma once
+
+#ifndef DEVICES_MACHINE_74259_H
+#define DEVICES_MACHINE_74259_H
+
+//**************************************************************************
+//  CONFIGURATION MACROS
+//**************************************************************************
+
+#define MCFG_ADDRESSABLE_LATCH_Q0_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 0, DEVCB_##_devcb);
+#define MCFG_ADDRESSABLE_LATCH_Q1_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 1, DEVCB_##_devcb);
+#define MCFG_ADDRESSABLE_LATCH_Q2_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 2, DEVCB_##_devcb);
+#define MCFG_ADDRESSABLE_LATCH_Q3_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 3, DEVCB_##_devcb);
+#define MCFG_ADDRESSABLE_LATCH_Q4_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 4, DEVCB_##_devcb);
+#define MCFG_ADDRESSABLE_LATCH_Q5_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 5, DEVCB_##_devcb);
+#define MCFG_ADDRESSABLE_LATCH_Q6_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 6, DEVCB_##_devcb);
+#define MCFG_ADDRESSABLE_LATCH_Q7_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_q_out_cb(*device, 7, DEVCB_##_devcb);
+
+#define MCFG_ADDRESSABLE_LATCH_PARALLEL_OUT_CB(_devcb) \
+	devcb = &addressable_latch_device::set_parallel_out_cb(*device, DEVCB_##_devcb);
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> addressable_latch_device
+
+class addressable_latch_device : public device_t
+{
+public:
+	// construction/destruction
+	addressable_latch_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock);
```

Comment:
As you might be able to tell, I started working on this device before the great device type encapsulation.